CompassSDK/Tracker/Core/CompassTracker.swift

```diff
@@ -310,24 +312,26 @@ extension CompassTracker: ConversionsProvider {
 }
 
 internal extension CompassTracker {
-    func getTrackingData(_ completion: @escaping (IngestTrackInfo) -> ()) {
-        getScrollPercent { [self] (scrollPercent) in
-            getConversions { [self] (conversions) in
-                if ((scrollPercent ?? 0) > (self.trackInfo.scrollPercent ?? 0)) {
-                    self.trackInfo.scrollPercent = scrollPercent
-                }
-                
-                var finalTrackInfo = self.trackInfo
-             
-                finalTrackInfo.conversions = conversions.isEmpty ? nil : conversions
-                finalTrackInfo.userVars = storage.userVars
-                finalTrackInfo.sessionVars = storage.sessionVars
-                finalTrackInfo.pageVars = pageVars
-                finalTrackInfo.userSegments = storage.userSegments
-                finalTrackInfo.hasConsent = storage.hasConsent
-                
-                completion(finalTrackInfo)
-           }
+    func getTrackingData(for conversion: String? = nil, tick: Int? = 0, _ completion: @escaping (IngestTrackInfo) -> ()) {
```

Comment:
- pass down converson to getTrackingData instead of grabbing all of them, so a task per conversion can be generated at the same time
- rely on dispatchGroup instead of operation finish for scheduling next tick, that is to say, if 4 ticks are generated because of 4 pending conversions, a single next tik should be scheduled, not 4.
- take tick management out of track info so I can increase it at any point, avoiding async race conditions.